cardinal/ecs/world.go

```diff
@@ -109,34 +110,24 @@ func (w *World) AddSystems(s ...System) {
 	}
 }
 
-// RegisterComponents attempts to initialize the given slice of components with a WorldAccessor.
-// This will give components the ability to access their own data.
-func (w *World) RegisterComponents(components ...component.IComponentType) error {
-	if w.stateIsLoaded {
+func RegisterComponent[T component.IAbstractComponent](world *World) error {
```

Comment:
hmm now that i think about it, should we just panic on errors in this function instead of returning an error? i imagine the register code for a game with lots of components will look quite hellish

Reply:
If we do this I'll make it another PR because this will modify a ton of tests. 

I think it's fine.